src/pk-dbus.c

```diff
@@ -45,6 +45,66 @@ static gpointer pk_dbus_object = NULL;
 
 G_DEFINE_TYPE (PkDbus, pk_dbus, G_TYPE_OBJECT)
 
+gboolean
+pk_dbus_get_uid_pid (PkDbus *dbus, const gchar *sender, guint32 *uid, guint32 *pid)
+{
+	GVariant *reply_var;
+	const char *key;
+	GVariant *value;
+	gboolean have_pid, have_uid;
+	g_autoptr(GVariantIter) iter = NULL;
+	g_autoptr(GError) error = NULL;
+
+	g_return_val_if_fail (PK_IS_DBUS (dbus), G_MAXUINT);
+	g_return_val_if_fail (sender != NULL, G_MAXUINT);
+
+	/* set in the test suite */
+	if (g_strcmp0 (sender, ":org.freedesktop.PackageKit") == 0) {
+		g_debug ("using self-check shortcut");
+		if (uid != NULL)
+			*uid = 500;
+		if (pid != NULL)
+			*pid = 12345;
+		return TRUE;
+	}
+
+	/* no connection to DBus */
+	if (dbus->priv->proxy_pid == NULL)
+		return FALSE;
+
+	/* get caller credentials from DBus */
+	reply_var = g_dbus_proxy_call_sync (dbus->priv->proxy_pid,
+					"GetConnectionCredentials",
+					g_variant_new ("(s)",
+						       sender),
+					G_DBUS_CALL_FLAGS_NONE,
+					2000,
+					NULL,
+					&error);
+	if (reply_var == NULL) {
+		g_warning ("Failed to get uid/pid for %s: %s",
+			   sender, error->message);
+		return G_MAXUINT;
+	}
+
+	have_pid = pid == NULL;
+	have_uid = uid == NULL;
```

Comment:
It seems confusing to call a variable `have_pid` and then assign it to true iff `pid` is a `NULL` pointer.